pkg/utils/secretsUtils.go

```diff
@@ -15,49 +15,44 @@ func (expDetails *ExperimentDetails) PatchSecrets(clients ClientSets, engineDeta
 		return err
 	}
 
-	log.Infof("Validating secrets specified in the ChaosExperiment & ChaosEngine")
-	err = expDetails.ValidateSecrets(clients)
-	if err != nil {
-		return err
+	if len(expDetails.Secrets) != 0 {
+		log.Infof("Validating secrets specified in the ChaosExperiment & ChaosEngine")
+		if err = expDetails.ValidateSecrets(clients); err != nil {
+			return err
+		}
 	}
 	return nil
 }
 
-// ValidateSecrets validates the secrets in application Namespace
-func (clientSets ClientSets) ValidateSecrets(secretName string, experiment *ExperimentDetails) error {
-
-	_, err := clientSets.KubeClient.CoreV1().Secrets(experiment.Namespace).Get(secretName, metav1.GetOptions{})
-	if err != nil {
-		return err
-	}
-	return nil
+// ValidateSecret validates the secret in Chaos Namespace
+func (clientSets ClientSets) ValidateSecret(secretName, namespace string) error {
+	_, err := clientSets.KubeClient.CoreV1().Secrets(namespace).Get(secretName, metav1.GetOptions{})
+	return err
 }
 
 // SetSecrets sets the value of secrets in Experiment Structure
 func (expDetails *ExperimentDetails) SetSecrets(clients ClientSets, engineDetails EngineDetails) error {
-	experimentSecrets, err := getExperimentSecrets(clients, expDetails)
+	experimentSecrets, err := expDetails.getExperimentSecrets(clients)
 	if err != nil {
 		return err
 	}
-	engineSecrets, err := getEngineSecrets(clients, engineDetails, expDetails)
+	engineSecrets, err := expDetails.getEngineSecrets(clients, engineDetails)
 	if err != nil {
 		return err
 	}
-
 	// Overriding the Secrets from the ChaosEngine
-	OverridingSecrets(experimentSecrets, engineSecrets, expDetails)
+	expDetails.OverridingSecrets(experimentSecrets, engineSecrets)
 
 	return nil
 }
 
-// ValidateSecrets checks for secrets in the Applicaation Namespace
+// ValidateSecrets checks for secrets in the Chaos Namespace
 func (expDetails *ExperimentDetails) ValidateSecrets(clients ClientSets) error {
-
 	for _, v := range expDetails.Secrets {
 		if v.Name == "" || v.MountPath == "" {
 			return errors.Errorf("Incomplete Information in Secret, will skip execution")
 		}
-		err := clients.ValidateSecrets(v.Name, expDetails)
+		err := clients.ValidateSecret(v.Name, expDetails.Namespace)
```

Comment:
To make it more explicit can we call it `ValidatePresenceOfSecretResource` in cluster